### PR TITLE
Make all mail parameters configurable

### PR DESCRIPTION
**Pull request by gumulka**
It is now possible to change the port and sender configurations for all instances, with still having reasonable defaults.

### CODE REVIEW

File: codalab/lib/codalab_manager.py [hunk 1]
```diff
@@ -395,9 +395,10 @@ def emailer(self):
                 host=self.config['email']['host'],
                 user=self.config['email'].get('user', 'noreply@codalab.org'),
                 password=self.config['email'].get('password', None),
-                use_tls=True,
-                default_sender='CodaLab <noreply@codalab.org>',
-                server_email='noreply@codalab.org',
+                use_tls=self.config['email'].get('use_tls',True),
+                port=self.config['email'].get('port', 587),
+                default_sender=self.config['email'].get('default_sender','CodaLab <noreply@codalab.org>'),
```
Comment (percyliang) [inline codalab/lib/codalab_manager.py]: Could we change the config keys:
- `default_sender` -> `sender_from`
- `server_email` -> `sender_email`
to make it semantically more clear?

**Review by percyliang — APPROVED** [review]
Thanks!  Just a small request for naming change.

**Comment by epicfaace** [comment]
Do we already have config parameters documented? If so, should probably update the documentation.

File: codalab/lib/codalab_manager.py (hunk 1 above)
Comment (gumulka) [reply codalab/lib/codalab_manager.py]: Sure. and already done

**Comment by gumulka** [comment]
> Do we already have config parameters documented? If so, should probably update the documentation.

No, you don't. See #1314

**Pull request merged**